test/lib/specifications/Specification.js

```diff
@@ -299,3 +308,88 @@ test("Invalid specVersion", async (t) => {
 		"For details, see https://sap.github.io/ui5-tooling/pages/Configuration/#specification-versions"
 	}, "Threw with expected error message");
 });
+
+test("getRootReader: Default parameters", async (t) => {
+	// Since Specification#create instantiates a far-away subclass, it would be a mess to mock
+	// every class up to "Specification.js" just to stub the resourceFactory's createReader method
+	// Therefore we just come up with our own subclass that can be instantiated right away:
+
+	const createReaderStub = sinon.stub();
+	const Specification = await esmock("../../../lib/specifications/Specification.js", {
+		"@ui5/fs/resourceFactory": {
+			createReader: createReaderStub
+		}
+	});
+
+	class DummySpecification extends Specification {
+		getPath() {
+			return "path";
+		}
+		getType() {
+			return "type";
+		}
+		getKind() {
+			return "kind";
+		}
+		getName() {
+			return "name";
+		}
+	}
+
+	const spec = new DummySpecification();
+	await spec.getRootReader();
+
+	t.is(createReaderStub.callCount, 1, "createReader got called once");
+	t.deepEqual(createReaderStub.getCall(0).args[0], {
+		fsBasePath: "path",
+		name: "Root reader for type kind name",
+		useGitignore: true,
+		virBasePath: "/",
+	}, "createReader got called with expected arguments");
+});
+
+test("getRootReader: Custom parameters", async (t) => {
+	const createReaderStub = sinon.stub();
+	const Specification = await esmock("../../../lib/specifications/Specification.js", {
+		"@ui5/fs/resourceFactory": {
+			createReader: createReaderStub
+		}
+	});
+
+	class DummySpecification extends Specification {
```

Comment:
dummy specification could be defined only once on top-level in the test file. so its' not necessary to define it twice.

Reply:
Not in this case, since it's extending a mocked `Specification` reference, with a `createReader` stub.

Reply:
true...missed that one. but could be extracted to a separate function where the test context is given and this function can be called from the test need this dummyspec

Reply:
Done 👌